### ISSUE TITLE
Query: Flatten out subquery models at query optimizer

### ISSUE THREAD

**Issue by smitpatel** [aspnet/EntityFramework#7613]
```
  'from Customer c in 
      from Customer c in DbSet<Customer>
      where c.City == "Seattle"
      order by c.CustomerID asc
      select c
  where c.CustomerID == "ALFKI"
  select c'
```

Query model like above can be optimized by merging the bodyclauses.
Points to consider:
- Inner selector is selecting query source directly.
- No multiple from clauses inside subquery
- Presently ReLinq blocks lifting of subquery if there is orderby clause due to possible ambiguity in ordering. We can still lift if only one (inner or outer) contains OrderBy clause.
- outer selector can be anything.
- No result operator on subquery.

**Comment by BladeWise** [comment]
Would this scenario be covered by such optiimization?

``` C#
 var entities = context.Set<Entity>();

var info = entities.Select(x => new
                                {
                                   MaxId = entities.Max(x => x.Id),
                                   Count = entities.Count()
                                })
                   .FirstOrDefault();
```

In this case the generated query is
``` SQL
SELECT TOP 1
     (SELECT MAX(ID) FROM Entities),
     (SELECT COUNT(*) FROM Entities)
FROM Entities
```
while the optimal result would be

``` SQL
SELECT TOP 1
     MAX(ID),
     COUNT(*)
FROM Entities
```

**Comment by smitpatel** [comment]
@maumar - Assigning this to you as this is lifting of order by.

**Comment by maumar** [comment]
fixed in f022d66bc3a461851a6250536399cd3b8f8bf8d2
